datafusion-examples/examples/parquet_sql_multiple_files.rs

```diff
@@ -33,13 +33,11 @@ async fn main() -> Result<()> {
 
     // Configure listing options
     let file_format = ParquetFormat::default().with_enable_pruning(true);
-    let listing_options = ListingOptions {
-        file_extension: FileType::PARQUET.get_ext(),
-        format: Arc::new(file_format),
-        table_partition_cols: vec![],
-        collect_stat: true,
-        target_partitions: 1,
-    };
+    let listing_options = ListingOptions::new(Arc::new(file_format))
+        .with_file_extension(FileType::PARQUET.get_ext())
+        .with_table_partition_cols(vec![])
+        .with_collect_stat(true)
+        .with_target_partitions(1);
```

Comment:
We could further  simplify this code and avoid explicitly specifying the default values (last three lines)

But that can also be done as a follow on PR